test/e2e/set/nodeconfig/nodeconfig_optimizations.go

```diff
@@ -0,0 +1,264 @@
+		}
+	})
+
+	g.It("should correctly project state for each scylla pod", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		nc := ncTemplate.DeepCopy()
+
+		g.By("Blocking node tuning")
+		// We have to make sure the namespace exists.
+		_, err := f.KubeAdminClient().CoreV1().Namespaces().Create(
+			ctx,
+			&corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: naming.ScyllaOperatorNodeTuningNamespace,
+				},
+			},
+			metav1.CreateOptions{},
+		)
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+
+		rq, err := f.KubeAdminClient().CoreV1().ResourceQuotas(naming.ScyllaOperatorNodeTuningNamespace).Create(
+			ctx,
+			&corev1.ResourceQuota{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "blocking",
+				},
+				Spec: corev1.ResourceQuotaSpec{
+					Hard: corev1.ResourceList{
+						corev1.ResourcePods: resource.MustParse("0"),
+					},
+				},
+			},
+			metav1.CreateOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Creating a NodeConfig")
+		nc, err = f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs().Create(ctx, nc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
+
+		framework.By("Creating a ScyllaCluster")
+		err = framework.SetupScyllaClusterSA(ctx, f.KubeClient().CoreV1(), f.KubeClient().RbacV1(), f.Namespace(), sc.Name)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for a ConfigMap to indicate blocking NodeConfig")
+		ctx2, ctx2Cancel := context.WithTimeout(ctx, apiCallTimeout)
+		defer ctx2Cancel()
+		podName := fmt.Sprintf("%s-%d", naming.StatefulSetNameForRack(sc.Spec.Datacenter.Racks[0], sc), 0)
+		pod, err := utils.WaitForPodState(ctx2, f.KubeClient().CoreV1(), sc.Namespace, podName, func(p *corev1.Pod) (bool, error) {
+			return true, nil
+		}, utils.WaitForStateOptions{})
+		cmName := naming.GetTuningConfigMapNameForPod(pod)
+		ctx3, ctx3Cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer ctx3Cancel()
+		src := &internalapi.SidecarRuntimeConfig{}
+		cm, err := utils.WaitForConfigMapState(ctx3, f.KubeClient().CoreV1().ConfigMaps(sc.Namespace), cmName, utils.WaitForStateOptions{}, func(cm *corev1.ConfigMap) (bool, error) {
+			if cm.Data == nil {
+				return false, nil
+			}
+
+			srcData, found := cm.Data[naming.ScyllaRuntimeConfigKey]
+			if !found {
+				return false, nil
+			}
+
+			err = json.Unmarshal([]byte(srcData), src)
+			if err != nil {
+				return false, err
+			}
+
+			return len(src.BlockingNodeConfigs) > 0, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(src.ContainerID).To(o.BeEmpty())
+		o.Expect(src.MatchingNodeConfigs).NotTo(o.BeEmpty())
+
+		waitTime := utils.RolloutTimeoutForScyllaCluster(sc) / 100 // FIXME
```

Comment:
FIXME - maybe check that port 10000 is closed - it's bound only to localhost so we would have to check it from inside the pod. Or 10001 port is opened when agent running inside the same pod is able to connect to it. So it should be closed too.

Reply:
I think the issue here is that to make sure it won't come up, we just have to wait.

Fixed the FIXME.